tests/Wallabag/CoreBundle/Controller/TagControllerTest.php

```diff
@@ -141,6 +141,53 @@ public function testRemoveTagFromEntry()
         $this->assertNull($tag, $this->tagName . ' was removed because it begun an orphan tag');
     }
 
+    public function testRemoveTag()
+    {
+        $this->logInAs('admin');
+        $client = $this->getClient();
+
+        $tag = new Tag();
+        $tag->setLabel($this->tagName);
+
+        $entry = new Entry($this->getLoggedInUser());
+        $entry->setUrl('http://0.0.0.0/foo');
+        $entry->addTag($tag);
+        $this->getEntityManager()->persist($entry);
+
+        $entry2 = new Entry($this->getLoggedInUser());
+        $entry2->setUrl('http://0.0.0.0/bar');
+        $entry2->addTag($tag);
+        $this->getEntityManager()->persist($entry2);
+        $this->getEntityManager()->flush();
+        $this->getEntityManager()->clear();
+
+        $client->request('GET', '/tag/delete/' . $tag->getSlug());
```

Comment:
@nicosomb wouldn't it be better to use a POST request for an action that alters the database?

Reply:
You're right ... for the API side. 

But here, it's not the API part. Our tests are web test cases. We just simulate a click on the link.

Reply:
I think even in interfaces it should be a POST action :)
using a form and a submit button.
IMO it's a good practice the have all actions that alter the database to be POST actions

Reply:
Why don't you use the client and then the crawler to retrieve the button and simulate a click on it?

Reply:
@j0k3r done